packages/react-reconciler/src/ReactFiber.js

```diff
@@ -443,7 +439,18 @@ export function createWorkInProgress(
   workInProgress.memoizedProps = current.memoizedProps;
   workInProgress.memoizedState = current.memoizedState;
   workInProgress.updateQueue = current.updateQueue;
-  workInProgress.dependencies = current.dependencies;
+
+  // Clone the dependencies object. This is mutated during the render phase, so
+  // it cannot be shared with the current fiber.
+  const currentDependencies = current.dependencies;
+  workInProgress.dependencies =
+    currentDependencies === null
+      ? null
+      : {
+          expirationTime: currentDependencies.expirationTime,
+          firstContext: currentDependencies.firstContext,
+          events: currentDependencies.events,
+        };
```

Comment:
The reason this only became an issue now is because of this change: https://github.com/facebook/react/pull/15944/files#diff-ff9ee977e81fe580b4202130fd86caeaR322

It used to reset `workInProgress.contextDependencies`, which is fine because `workInProgress` is a clone. But after #15944 it resets `workInProgress.dependencies.firstContext`, which is a bug because `dependencies is shared with the current fiber.

I clone in the render phase instead, but since this is such a subtle thing to keep track of (I've hit a similar thing with the class update queue before, but even so it took me a while to figure this bug out), I opted to clone here instead. Something to keep in mind when we look at refactoring the Fiber data structure.